### PR TITLE
models and types

### PR DESCRIPTION
**Pull request by joelwass**
Added models and types for sequelize

### CODE REVIEW

File: src/@types/donation.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface Donation {
+  id: number,
+  amount: number,
+  frequency: string,
+  meta: any
+}
+
+export enum Frequency {
+  Monthly = 'monthly',
```
Comment (stripedpajamas) [inline src/@types/donation.ts]: Can you make case consistent in this enum?

File: src/models/account.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          })
+      }
+    }
+  })
+
+  Account.authenticate = (body: any) => {
+    let account: any
+    return Account.findOne({ where: { email: body.email } })
+      .then((localAccount: UserAccount) => {
+        if (!localAccount) return Promise.reject(new helper.CustomError(helper.strings.sorryWeCantFindEmail))
```
Comment (stripedpajamas) [inline src/models/account.ts]: We don't want to leak valid emails; invalid username/password should be the generic message for failures in authentication

File: src/@types/user.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface User {
```
Comment (stripedpajamas) [inline src/@types/user.ts:1]: What's the difference between UserAccount and User? Is the separation useful?

File: src/controllers/teacher/index.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import {BaseContext} from 'koa'
+import * as Strings from '../../helpers/strings'
+import * as teacherController from './teacher'
+import {Teacher} from '../../@types/teacher'
+
+export default class TeacherController {
+  constructor() {}
```
Comment (stripedpajamas) [inline src/controllers/teacher/index.ts]: Unneeded line here

File: src/controllers/teacher/index.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    ctx.assert(email, 400, Strings.EmailIsRequired)
+
+    const teacher: Teacher = await teacherController.getTeacher(email)
+
+    ctx.status = 200
+    ctx.body = { ok: true, teacher }
+  }
+
+  public async getAllTeachers(ctx: BaseContext) {
+    const teachers: Teacher[] = await teacherController.getAllTeachers()
```
Comment (stripedpajamas) [inline src/controllers/teacher/index.ts:36]: More to be done on this route, right? Add a `TODO` if so, so we don't forget. Currently gets teachers and returns nothing.

File: src/models/account.ts (hunk 2 above)
Comment (joelwass) [reply src/models/account.ts]: so not user friendly. w/e. will change

File: src/@types/user.ts (hunk 3 above)
Comment (joelwass) [reply src/@types/user.ts:1]: a UserAccount is a user (donor) who has an account. 
A TeacherAccount is a teacher who has an account.
a User is just someone who wants to subscribe or stay involved to SOME capacity (only required field is email).

Comment (joelwass) [reply src/@types/user.ts:1]: resolving this.

**Pull request merged**